### PR TITLE
flexoelectricity mixed formulation terms

### PR DESCRIPTION
**Pull request by rc**

### CODE REVIEW

File: sfepy/terms/terms_flexo.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
+"""
+Flexoelectricity related terms.
+"""
+import numpy as nm
+
+from sfepy.mechanics.tensors import dim2sym
+from sfepy.terms.terms_multilinear import ETermBase
+
+def make_grad2strain(dim):
```
Comment (VascoSch92) [inline sfepy/terms/terms_flexo.py:9]: what happens if `dim` is not 1, 2, or 3?

File: sfepy/terms/terms_flexo.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+    def get_function(self, mat, tvar, svar, mode=None, term_mode=None,
+                     diff_var=None, **kwargs):
+        dim = self.region.dim
+        sym = dim2sym(dim)
+        aux = make_grad2strain(dim)
+        n_qp = mat.shape[1]
+        mat = mat.reshape((-1, n_qp, dim, dim, sym))
+
+        if term_mode is None:
```
Comment (VascoSch92) [inline sfepy/terms/terms_flexo.py:133]: what if `term_mode` is not one of 'electric_displacement',  'double_stress' or `None` ?

File: sfepy/terms/terms_flexo.py (hunk 1 above)
Comment (rc) [reply sfepy/terms/terms_flexo.py:9]: Thanks, fixed!

File: sfepy/terms/terms_flexo.py (hunk 2 above)
Comment (rc) [reply sfepy/terms/terms_flexo.py:140]: Thanks, fixed!

**Comment by rc** [comment]
The CI failures are due to numpy 2.0.0 - I forgot to pin the version to <2.

**Pull request merged**